optuna/_callbacks.py

```diff
@@ -55,3 +55,56 @@ def __call__(self, study: "optuna.study.Study", trial: FrozenTrial) -> None:
         n_complete = len(trials)
         if n_complete >= self._n_trials:
             study.stop()
+
+
+@experimental("2.8.0")
+class RetryFailedTrialCallback:
```

Comment:
For maintainers, it might be a bit confusing having defined this class in `optuna/_callbacks.py`, given its API namespace under `storages`. How about leaving a TODO comment to potentially move the implementation to `storages/` as well?

Reply:
Not sure about this one. I thought the idea of having an `optuna/_callbacks.py` file was to put all callbacks there. If this is better put in the `storages/` then I can move it there now. 🤷‍♂️

Reply:
It's an implementation details so it's not a strong opinion. Actually, `MaxTrialsCallback` is exposed via `optuna.study` and not `optuna` so it's already a bit confusing and maybe we can leave it as is. Was first thinking of something along `storages/_callbacks.py`.